internal/resources/aws/transfer_server.go

```diff
@@ -0,0 +1,138 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/shopspring/decimal"
+
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+)
+
+// TransferServer defines a AWS Transfer Server resource from Transfer Family
+// service. It supports multiple transfer protocols like FTP/FTPS/SFTP and
+// each is billed hourly when enabled. It also bills the amount of data
+// being downloaded/uploaded over those protocols.
+//
+// See more resource information here: https://aws.amazon.com/aws-transfer-family/.
+//
+// See the pricing information here: https://aws.amazon.com/aws-transfer-family/pricing/.
+type TransferServer struct {
+	Address   string
+	Region    string
+	Protocols []string
+
+	// "usage" args
+	MonthlyDataDownloadedGB *float64 `infracost_usage:"monthly_data_downloaded_gb"`
+	MonthlyDataUploadedGB   *float64 `infracost_usage:"monthly_data_uploaded_gb"`
+}
+
+// TransferServerUsageSchema defines a list of usage items for TransferServer.
+var TransferServerUsageSchema = []*schema.UsageItem{
+	{Key: "monthly_data_downloaded_gb", DefaultValue: 0, ValueType: schema.Float64},
+	{Key: "monthly_data_uploaded_gb", DefaultValue: 0, ValueType: schema.Float64},
+}
+
+// PopulateUsage parses the u schema.UsageData into the TransferServer.
+// It uses the `infracost_usage` struct tags to populate data into the TransferServer.
+func (t *TransferServer) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(t, u)
+}
+
+// BuildResource builds a schema.Resource from a valid TransferServer struct.
+// This method is called after the resource is initialised by an IaC provider.
+func (t *TransferServer) BuildResource() *schema.Resource {
+	costComponents := []*schema.CostComponent{}
+
+	service := "AWSTransfer"
+	productFamily := "AWS Transfer Family"
+
+	for _, protocol := range t.Protocols {
+		costComponents = append(costComponents,
+			newProtocolCostComponent(
+				protocol,
+				"[A-Z0-9]*-ProtocolHours",
+				t.Region,
+				service,
+				productFamily,
+			),
+		)
+	}
+
+	costComponents = append(costComponents,
+		newDataTransferCostComponent(
+			"Data downloaded",
+			t.MonthlyDataDownloadedGB,
+			"[A-Z0-9]*-DownloadBytes",
+			t.Region,
+			service,
+			productFamily,
+		),
+		newDataTransferCostComponent(
+			"Data uploaded",
+			t.MonthlyDataUploadedGB,
+			"[A-Z0-9]*-UploadBytes",
+			t.Region,
+			service,
+			productFamily,
+		),
+	)
+
+	return &schema.Resource{
+		Name:           t.Address,
+		UsageSchema:    TransferServerUsageSchema,
+		CostComponents: costComponents,
+	}
+}
+
+func newProtocolCostComponent(protocol string, usageType string, region string, service string, productFamily string) *schema.CostComponent {
```

Comment:
Should we have this function as a method on the struct so it doesn't pollute the aws package?

Reply:
Yes, totally! Thanks for catching! I forgot that the full package shares the functions 😅. Please see the fixup commit with some refactoring.